model/src/main/java/org/projectnessie/model/ContentKey.java

```diff
@@ -91,7 +91,7 @@ protected void validate() {
         throw new IllegalArgumentException(
             String.format("Content key '%s' must not contain a null element.", elements));
       }
-      if (e.contains(ZERO_BYTE_STRING)) {
+      if (e.contains(ZERO_BYTE_STRING) || e.contains(ZERO_BYTE_STRING_URL_ENCODED)) {
```

Comment:
Having two different representations for a zero-char is IMO not good.
The content-keys `a.f\u0000o.bar` and `a.f%00o.bar` are technically different.
However, for this error, not sure whether we need it, because `%` should already be forbidden IIRC.

Reply:
My reading of this PR is that now, `a.f%00o.bar` is the canonical representation. `a.f\u0000o.bar` is supported for backward compatibility.

Reply:
`%` isn't forbidden btw

Reply:
see also my above comment about having a single canonical representation

Reply:
We reuse the URLEncoder for escaping null chars, but the `toPathString` form is not quite the same as the actual URL form. For example, `a` > `f.o` > `bar` will have the `toPathString` form of `a.f%00o.bar` but in a URL it will be `a.f%2500o.bar`.

So, essentially, we will have double URL encoding. If this is too confusing, what do you think about using some non-ascii unicode char instead of `\u0000`, e.g. `\u203B` (reference mark) https://www.unicode.org/charts/PDF/U2000.pdf